e2e/main.sh

```diff
@@ -107,6 +107,30 @@ apiServer:
   subjectAltNames:
   - '"${USHIFT_IP}"'
 EOF' &>"${output_dir}/0001-setup.log"
+
+    ssh_cmd 'cat <<EOF | sudo tee /etc/microshift/lvmd.yaml
+    # The fields here reflect LVM state of the CI smoke-testing instance.
+    device-classes:
+      # The name of a device-class
+      - name: default
+
+        # The group where this device-class creates the logical volumes
+        volume-group: rhel
+
+        # Storage capacity in GiB to be spared
+        spare-gb: 0
+
+        # A flag to indicate that this device-class is used by default
+        default: true
+      - name: thin
```

Comment:
Should we add this to the defaults used if the user does not give a config file?

Reply:
I was wondering that too.  We'd also need to add additional checks for thin volume pools b/c thin provisioning doesn't directly consume volume groups.

Reply:
So the user has to do something additional on the system to enable thin volume pools before they can use this feature?

Reply:
Correct - it adds an additional install step to create the thin-volume pool on the volume group.

Reply:
It sounds like we probably should not provide a default then. Perhaps in a future version we could add a default if we can detect a valid volume pool, but we want to wrap up the first iteration of this ASAP so let's not block on that.

What happens if the user does not provide any of the backup-related config at all (thin device class, any CRs needed, etc.) but all of the controllers are running? Is something going to throw an error or would it just fail to process their request for a snapshot because there's nowhere to put it?

Reply:
The controllers will run normally, but won't be able to fulfill snapshot/restore requests.  The errors are recorded as events on the user's object.  The webhook alleviates some of the debugging for the user.  It will refuse snapshots for PVCs that don't exists, and similarly for PVCs that restore from snapshots that don't exist.

Reply:
OK, that seems like a good place to start. Once we have it working, we could improve it to not run the snapshot controller at all if there is no storage set up for it, but let's take that as a future enhancement and not hold up the first version.